### ISSUE TITLE
Invalid import of PIL.Image in sphinx/writers/html.py

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#920]
There is a code block in sphinx/writers/html.py (checked with Sphinx 1.1.3 and /trunk sources):

```
#!python
try:
    from PIL import Image        # check for the Python Imaging Library
except ImportError:
    Image = None
```

The problem with that block is that on many systems (where PIL is installed as site-package), there is no such "module" as "PIL" (even though PIL is installed). I.e. "import Image" will work, but "import PIL" will not.

I see this problem on Windows with Python 2.7:

```
#!python
>>> __import__('PIL')
ImportError: No module named PIL
>>> __import__('Image')
<module 'Image' from 'C:\Python27\lib\site-packages\pil-1.1.7-py2.7-win32.egg\Image.pyc'>
```

As well as on RHEL6 with Python 2.6:

```
#!python
>>> __import__('PIL')
ImportError: No module named PIL
>>> __import__('Image')
<module 'Image' from '/usr/lib/python2.6/site-packages/PIL-1.1.7-py2.6-linux-i686.egg/Image.pyc'>
```

However, it works just fine on Ubuntu 12, where PIL is a dist-package named "PIL":

```
#!python
>>> __import__('PIL')
<module 'PIL' from '/usr/lib/python2.7/dist-packages/PIL/__init__.pyc'>
>>> __import__('Image')
<module 'Image' from '/usr/lib/python2.7/dist-packages/PIL/Image.pyc'>
```

The point is, it all works well only when PIL installation folder is called "PIL" which doesn't happen that often.

So I'd propose either to fix this check the following simple way:

```
#!python
try:
    import Image # just import Image - it will be PIL.Image if it's installed
except ImportError:
    Image = None
```

or in a more complicated way like this:

```
#!python
# Keep this import - try to import PIL.Image this way first
try:
    from PIL import Image        # check for the Python Imaging Library
except ImportError:
    Image = None

# And if it failed - try in a more complicated way
if not Image:
    try:
        # Trying to import Image directly, to handle cases when PIL
        # is installed to site-package which is not called 'PIL'
        # and check that signature of the imported Image match
        # http://www.pythonware.com/library/pil/handbook/image.htm
        Image = __import__('Image')
        if not ('blend' and 'composite' and 'frombuffer') in dir(i):
            Image = None
    except ImportError:
        Image = None
```

Some references to the same problem:
- http://stackoverflow.com/questions/8339991/why-in-python-sometimes-from-pil-import-image-fails-and-import-image-works
- https://code.djangoproject.com/ticket/6054

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/920
- Originally reported by: [MaxKuzkin](https://bitbucket.org/MaxKuzkin)
- Originally created at: 2012-05-04T12:26:45.212

**Comment by shimizukawa** [comment]
_From [MaxKuzkin](https://bitbucket.org/MaxKuzkin) on 2012-05-04 10:40:55+00:00_

Patch that fixes this problem

**Comment by shimizukawa** [comment]
_From Marc Schlaich on 2012-05-08 12:46:00+00:00_

I can confirm this problem. But the correct solution would be IMO:

{{{
#!python
try:
    from PIL import Image 
except ImportError:
    try:
        import Image 
    except ImportError:
        Image = None
}}}

**Comment by shimizukawa** [comment]
_From [MaxKuzkin](https://bitbucket.org/MaxKuzkin) on 2012-05-14 06:44:28+00:00_

Yes, this fix works too. I'm just not 100% sure that in all configurations "Image" will be the right "Image" - which may cause even more problems at runtime. Though I've never seen any other "Image" as popular as from "PIL".
I will appreciate any fix in the repository - so that we don't have to patch distribution of Sphinx to make it work.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-06 06:17:11+00:00_

It is PIL's packaging issue.

Please use `Pillow` that is PIL but resolve some PIL's packaging issues.

Even so, I think [Marc Schlaich](https://bitbucket.org/schlamar) 's patch is reasonable for a while.

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2012-11-06 09:38:30+00:00_

Yes, I agree about using [Marc Schlaich](https://bitbucket.org/schlamar)'s patch.  This is also the "Image" that docutils imports so it will be used in other cases anyway.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-06 22:43:49+00:00_

Closes #920: rescue PIL packaging issue that allow import `Image` without `PIL` namespace.
Also closes #702.

→ <<cset 99621d7a01638872a5af727aaa8714b269c0f769>>

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-06 22:50:51+00:00_

I applied [Marc Schlaich](https://bitbucket.org/schlamar)'s patch to html writer.

BTW, [builders/epub.py](https://github.com/sphinx-doc/testing/blob/99621d7a01638872a5af727aaa8714b269c0f769/sphinx/builders/epub.py#L21)  already implemented the same logic.
